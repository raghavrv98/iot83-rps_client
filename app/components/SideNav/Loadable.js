/**
 *
 * Asynchronously loads the component for SideNav
 *
 */

import loadable from '@loadable/component';

export default loadable(() => import('./index'));
