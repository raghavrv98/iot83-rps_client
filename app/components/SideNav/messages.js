/*
 * SideNav Messages
 *
 * This contains all the text for the SideNav component.
 */

import { defineMessages } from "react-intl";

export const scope = "app.components.SideNav";

export default defineMessages({
  welcome: {
    id: `${scope}.welcome`,
    defaultMessage: "Welcome !"
  },
  changePassword: {
    id: `${scope}.changePassword`,
    defaultMessage: "Change Password"
  },
  logout: {
    id: `${scope}.logout`,
    defaultMessage: "Logout"
  },
  manageNavigations: {
    id: `${scope}.manageNavigations`,
    defaultMessage: "Manage Navigations"
  },
  managebranding: {
    id: `${scope}.managebranding`,
    defaultMessage: "Manage Branding"
  },
  managePasswordRequest: {
    id: `${scope}.managePasswordRequest`,
    defaultMessage: "Manage Password Request"
  },
  allAvailableAlarmsAndAlerts: {
    id: `${scope}.allAvailableAlarmsAndAlerts`,
    defaultMessage: "Alarms"
  }
});
