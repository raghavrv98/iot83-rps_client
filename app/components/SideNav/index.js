/**
 *
 * SideNav
 *
 */

import React from "react";
import { Link } from 'react-router-dom';
import { FormattedMessage } from "react-intl";
import LocaleToogle from '../../containers/LocaleToggle/Loadable'
import { logoutHandler } from '../../utils/commonUtils'
import messages from "./messages";
import commonMessages from '../../messages';


/* eslint-disable react/prefer-stateless-function */
class SideNav extends React.Component {

    state = {
        imageUrl: '../../assets/images/nVent-icon.png',
        navItems: [],
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.navItems && nextProps.navItems !== this.props.navItems) {
            let activePage = {};
            nextProps.navItems.map((val, index) => {
                activePage[val.navName] = []
                if (index === 0) {
                    activePage[val.navName].push("/");
                }
                val.subMenus.length > 0 ?
                    val.subMenus.map(valSub => { activePage[val.navName].push("/" + valSub.url); }) :
                    activePage[val.navName].push("/" + val.url)
            })
            this.setState({
                navItems: nextProps.navItems,
                activePage,
            })
        }
    }

    render() {
        return (
            <header className="appHeader">
                <nav className="navbar navbar-expand-lg">
                    <a className="navbar-brand">
                        <img
                            src={`${window.API_URL}api/public/static/logo/pmmp.jpeg`}
                            onError={(e) => { e.target.onerror = null; e.target.src = require('../../assets/images/nVent-icon.png') }}
                        />
                    </a>
                    <button className="navbar-toggler text-light" type="button" data-toggle="collapse" data-target="#navBarMob" aria-controls="navBarMob" aria-expanded="false" aria-label="Toggle navigation">
                        <i className="fas fa-bars"></i>
                    </button>
                    <div className="collapse navbar-collapse navbar-mob animated fadeInDown" id="navBarMob">
                        <ul className="navbar-nav">
                            {this.state.navItems.map(nav => (
                                nav.subMenus.length > 0 ?
                                    <li key={nav.navName} className={`nav-item dropdown nav-subItem ${this.state.activePage[nav.navName].includes(window.location.pathname) ? "active" : ""}`}>
                                        <a className="nav-link">
                                            {nav.navName}
                                            <i className="fal fa-angle-down" />
                                        </a>
                                        <ul className="dropdown-menu animated fadeInDown">
                                            {nav.subMenus.map(navSub => (<li className="dropdown-item" key={navSub.url}>
                                                <Link to={"/" + navSub.url} target={navSub.newTab ? "_blank" : ""}>
                                                    {navSub.navName}
                                                </Link>
                                            </li>))}
                                        </ul>
                                    </li>
                                    :
                                    <li key={nav.url} className={`nav-item ${this.state.activePage[nav.navName].includes(window.location.pathname) ? "active" : ""}`}>
                                        <Link to={"/" + nav.url} className="nav-link" target={nav.newTab ? "_blank" : ""}>
                                            {nav.navName}
                                        </Link>
                                    </li>
                            ))}
                        </ul>
                    </div>
                </nav>
                {this.props.verified === "true" ?
                    <ul className="appInfo">
                        <li className="userInfo userInfo-lg">
                            <p><FormattedMessage {...messages.welcome} children={(message => message)} /><strong>{this.props.username ? this.props.username : "Admin"}</strong></p>
                            <h6>( {this.props.role} )</h6>
                        </li>
                        <li>
                            {/* <LocaleToogle /> */}
                            <div className="LanguageSelected">
                                <span>EN</span>
                            </div>
                        </li>
                        <li className="options">
                            <div className="dropdown">
                                <button className="btn dropdown-toggle" data-toggle="dropdown">
                                    <i className="fal fa-cogs text-white"></i>
                                </button>
                                <div className="dropdown-menu animated rotateInUpRight">
                                    <div className="dropdown-item userInfo userInfo-md">
                                        <p><FormattedMessage {...messages.welcome} children={(message => message)} /> <strong>{this.props.username ? this.props.username : "Admin"}</strong></p>
                                        <h6>( {this.props.role} )</h6>
                                    </div>
                                    <div className="dropdown-item" id="changePassword" onClick={() => { this.props.history.push('/changePassword') }}><FormattedMessage {...messages.changePassword} children={(message => message)} /></div>
                                    <div className="dropdown-item" id="settings" onClick={() => { this.props.history.push('/settings') }}><FormattedMessage {...commonMessages.settings} children={(message => message)} /></div>
                                    {this.props.tenant ? (<div className="dropdown-item" id="workingSkeletonDemo" onClick={() => { this.props.history.push('/managePasswordRequest') }}><FormattedMessage {...messages.managePasswordRequest} children={(message => message)} /></div>) : null}
                                    <div className="dropdown-item" onClick={() => logoutHandler()}><FormattedMessage {...messages.logout} children={(message => message)} /></div>
                                </div>
                            </div>
                        </li>
                    </ul> :
                    <ul className="appInfo">
                        <li className="userInfo pd-r-20">
                            <p><FormattedMessage {...messages.welcome} children={(message => message)} /> <strong>{this.props.username ? this.props.username : "Admin"}</strong></p>
                            <h6>( {this.props.role} )</h6>
                        </li>
                    </ul>
                }
            </header>
        );
    }
}

SideNav.propTypes = {};

export default SideNav;
