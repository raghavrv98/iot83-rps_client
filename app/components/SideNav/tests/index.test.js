import React from 'react';
import { shallow } from 'enzyme';
import history from 'utils/history';

import SideNav from '../index';
const locationJson = {
  'dashboard' : "/",
  'iam' : "/manageUsers",
  'manageTenants' : "/manageTenants",
  'manageAgents' : "/manageAgents",
  'managePlant' : "/managePlant",
  'manageNavigation' : "/manageNavigation",
}
describe('<SideNav /> for verified user', () => {
  it('match snapshots', () => {
    const wrap = shallow(<SideNav tenant="guest" verified="true" location={{pathname: locationJson.manageNavigation}} />)
    expect(wrap).toMatchSnapshot()
  })

  it("test component will recieve props", async () => {
    spyOn(SideNav.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<SideNav history={history} verified="true" username="user" navItems={undefined}  />);
    wrapper.setProps({ navItems: [{navName: "sample", subMenus: []},
    {navName: "sample1", subMenus: [{url: ""}]}] });
    expect(SideNav.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  });

  // it("clicking logout trigger onClick event", async () => {
  //   const wrapper = shallow(<SideNav  tenant="guest" verified="true" history={history} />);
  //   let logout = wrapper.find("li.logout");
  //   expect(logout.length).toBe(1);
  //   logout.simulate('click')
  //   expect(history.location.pathname).toEqual("/");
  // });

  // test('calls walkingSkelton click', async () => {
  //   const wrapper = shallow(<SideNav tenant="guest" verified="true"  history={history} />)
  //   let walkingSkelton = wrapper.find("li#workingSkeletonDemo");
  //   expect(walkingSkelton.length).toBe(1);
  //   walkingSkelton.simulate('click')
  //   expect(history.location.pathname).toEqual("/managePasswordRequest");
  // });

  // test('calls Async Click function', async () => {
  //   const wrapper = shallow(<SideNav tenant="guest" verified="true" location={{pathname: locationJson.manageNavigation}}  history={history} />)
  //   wrapper.instance().clearLocalStorage();
  //   expect(history.location.pathname).toEqual("/login");
  // });

  // it("clicking change password trigger onClick event", async () => {
  //   const wrapper = shallow(<SideNav tenant="guest" verified="true" location={{pathname: locationJson.dashboard}} history={history} />);
  //   let changePassword = wrapper.find("li#changePassword");
  //   expect(changePassword.length).toBe(1);
  //   changePassword.simulate('click')
  //   expect(history.location.pathname).toEqual("/changePassword");
  // });

  // it("clicking settings trigger onClick event", async () => {
  //   const wrapper = shallow(<SideNav tenant="guest" verified="true" location={{pathname: locationJson.dashboard}} history={history} />);
  //   let changePassword = wrapper.find("li#settings");
  //   expect(changePassword.length).toBe(1);
  //   changePassword.simulate('click')
  //   expect(history.location.pathname).toEqual("/settings");
  // });
})

// describe('<SideNav /> for not verified user', () => {
//   it("check top nav", async () => {
//     const wrapper = shallow(<SideNav tenant="guest" verified="false" username="demo" />);
//     let topNavProfileDiv = wrapper.find("div.topNavProfile");
//     expect(topNavProfileDiv.exists()).toBeTruthy();
//     expect(topNavProfileDiv.children().length).toBe(1);
//   });
// })
// it("check top nav", async () => {
//   const wrapper = shallow(<SideNav tenant="guest" verified="false" />);
//   let topNavProfileDiv = wrapper.find("div.topNavProfile");
//   expect(topNavProfileDiv.exists()).toBeTruthy();
//   expect(topNavProfileDiv.children().length).toBe(1);
// });

