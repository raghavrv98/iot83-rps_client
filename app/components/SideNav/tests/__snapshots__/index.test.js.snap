// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav /> for verified user match snapshots 1`] = `
<header
  className="appHeader"
>
  <nav
    className="navbar navbar-expand-lg"
  >
    <a
      className="navbar-brand"
    >
      <img
        onError={[Function]}
        src="undefinedapi/public/static/logo/pmmp.jpeg"
      />
    </a>
    <button
      aria-controls="navBarMob"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler text-light"
      data-target="#navBarMob"
      data-toggle="collapse"
      type="button"
    >
      <i
        className="fas fa-bars"
      />
    </button>
    <div
      className="collapse navbar-collapse navbar-mob animated fadeInDown"
      id="navBarMob"
    >
      <ul
        className="navbar-nav"
      />
    </div>
  </nav>
  <ul
    className="appInfo"
  >
    <li
      className="userInfo userInfo-lg"
    >
      <p>
        <FormattedMessage
          defaultMessage="Welcome !"
          id="app.components.SideNav.welcome"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <strong>
          Admin
        </strong>
      </p>
      <h6>
        ( 
         )
      </h6>
    </li>
    <li>
      <div
        className="LanguageSelected"
      >
        <span>
          EN
        </span>
      </div>
    </li>
    <li
      className="options"
    >
      <div
        className="dropdown"
      >
        <button
          className="btn dropdown-toggle"
          data-toggle="dropdown"
        >
          <i
            className="fal fa-cogs text-white"
          />
        </button>
        <div
          className="dropdown-menu animated rotateInUpRight"
        >
          <div
            className="dropdown-item userInfo userInfo-md"
          >
            <p>
              <FormattedMessage
                defaultMessage="Welcome !"
                id="app.components.SideNav.welcome"
                values={Object {}}
              >
                <Component />
              </FormattedMessage>
               
              <strong>
                Admin
              </strong>
            </p>
            <h6>
              ( 
               )
            </h6>
          </div>
          <div
            className="dropdown-item"
            id="changePassword"
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Change Password"
              id="app.components.SideNav.changePassword"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </div>
          <div
            className="dropdown-item"
            id="settings"
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Settings"
              id="app.common.messages.settings"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </div>
          <div
            className="dropdown-item"
            id="workingSkeletonDemo"
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Manage Password Request"
              id="app.components.SideNav.managePasswordRequest"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </div>
          <div
            className="dropdown-item"
            onClick={[Function]}
          >
            <FormattedMessage
              defaultMessage="Logout"
              id="app.components.SideNav.logout"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
          </div>
        </div>
      </div>
    </li>
  </ul>
</header>
`;
